contributor-docs/events-notation.md

```diff
@@ -0,0 +1,141 @@
+# Event Notation Guide
+
+Concise notation for describing event sequences in Livestore.
+
+## Core Concepts
+
+### Event Sequence Numbers
+
+Events use integer sequence numbers and are denoted as `e0`, `e1`, `e2`, etc. These correspond to the `EventSequenceNumber` type (see `packages/@livestore/common/src/schema/EventSequenceNumber.ts`) which contains:
+- **global**: Globally unique integer sequence number
+- **client**: Client-only event counter (default: 0)
+- **rebaseGeneration**: Increments when client rebases (default: 0)
+
+### Notation Format
+
+```
+[Client:]e{global}[+{client}][r{rebaseGeneration}]['][{context}][(←origin)]
+```
+
+The base format `e{global}[+{client}][r{rebaseGeneration}]` is implemented by `EventSequenceNumber.toString()` and `EventSequenceNumber.fromString()`.
+
+Core examples:
+- `e0` - Global event 0
+- `e3'` - Unconfirmed event 3
+- `e5+1` - Client-only event (global: 5, client: 1)
```

Comment:
I would use `.` over `+` because:

  - **Hierarchical feeling** - Visually suggests e5.1 is "under" or "part of" e5
  - **Familiar from versioning** - 1.2.3 semantic versioning is widely understood
  - **Less visually dense** - More compact, easier to read in complex notations:
```
Dot:  B:e5.1'{user.update}(←e3')
Plus: B:e5+1'{user.update}(←e3')
```